### PR TITLE
Add in filter

### PR DESCRIPTION
**Pull request by archolewa**
We don't currently have a Filter for Druid's In Filter. I'd like to have one, so I've added it. 😄 

Note that we are not currently using it, though we should be able to use it to simplify the filter builders. 

I tried, but fixing the tests turned out to be taking more time than I can really afford right now. Especially if we want to make the tests readable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,10 +10,16 @@ Current
 
 ### Added:
 
+- [Support for Druid's In Filter](https://github.com/yahoo/fili/pull/64)
```
Comment (cdeszaq) [inline CHANGELOG.md]: Should add a note that it only works with v0.9.1 (I think... I don't think it was there in earlier versions)

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,10 +10,16 @@ Current
...
+
 - [Documentation for topN](https://github.com/yahoo/fili/pull/43)
 
 ### Changed:
 
+- [Cleaned up the type for `DimensionalFilter::withDimension`](https://github.com/yahoo/fili/pull/64)
+    -- Previously, `DimensionalFilter::withDimension` had the signature `public abstract <T> T withDimension(Dimension dimension)`,
+    but `T` is already defined as part of the class type of `DimensionalFilter`. Therefore, `<T>` has been removed.
```
Comment (cdeszaq) [inline CHANGELOG.md]: Does that actually change anything? If not (ie. if it's the same as changing `List<Fool> f = new ArrayList<Foo>()` into `List<Fool> f = new ArrayList<>()`), then I wouldn't bother noting it in the change log.

File: fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/InFilter.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.Objects;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Fili model of the Druid In Filter: http://druid.io/docs/0.9.1.1/querying/filters.html.
+ * <p>
+ * An In Filter is a generalization of the {@link SelectorFilter}. Rather than filtering on a specific value for a
+ * specific diimension, the In Filter takes a list of values. A dimension satisfies the In Filter iff its value is
+ * contained in the specified list.
```
Comment (cdeszaq) [inline fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/InFilter.java]: May be worth mentioning that an `in` filter is the same as an `or` filter around a collection of `selector` filters. (I think that's right, right?)

Comment (cdeszaq) [inline fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/InFilter.java]: Also should mention here that this is only supported v0.9.1 and later.

File: CHANGELOG.md (hunk 1 above)
Comment (archolewa) [inline CHANGELOG.md]: Looks like it is also in v0.9.0 (at least it's included in the docs...)

File: CHANGELOG.md (hunk 2 above)
Comment (archolewa) [inline CHANGELOG.md]: No, I don't think it changes anything.

File: fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/InFilter.java (hunk 3 above)
Comment (archolewa) [inline fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/InFilter.java]: Yes.

**Comment by michael-mclawhorn** [comment]
👍 
I don't disagree with Rick's input, but this is techincally sound.

File: CHANGELOG.md (hunk 1 above)
Comment (cdeszaq) [inline CHANGELOG.md]: If it's included, sweet. But that means what we should just indicate that it's available in v0.9.0 instead o v0.9.1 :-P

**Pull request merged**